src/legacy/core_plugins/data/public/expressions/expression_renderer.tsx

```diff
@@ -19,38 +19,41 @@
 
 import { useRef, useEffect } from 'react';
 import React from 'react';
-import { Ast } from '@kbn/interpreter/common';
 
-import { ExpressionRunnerOptions, ExpressionRunner } from './expression_runner';
-import { Result } from './expressions_service';
+import { ExpressionAST, IExpressionLoaderParams, IInterpreterResult } from './lib/_types';
+import { IExpressionLoader, ExpressionLoader } from './lib/loader';
 
 // Accept all options of the runner as props except for the
 // dom element which is provided by the component itself
-export type ExpressionRendererProps = Pick<
-  ExpressionRunnerOptions,
-  Exclude<keyof ExpressionRunnerOptions, 'element'>
-> & {
-  expression: string | Ast;
+export interface ExpressionRendererProps extends IExpressionLoaderParams {
+  expression: string | ExpressionAST;
   /**
    * If an element is specified, but the response of the expression run can't be rendered
    * because it isn't a valid response or the specified renderer isn't available,
    * this callback is called with the given result.
    */
-  onRenderFailure?: (result: Result) => void;
-};
+  onRenderFailure?: (result: IInterpreterResult) => void;
+}
 
 export type ExpressionRenderer = React.FC<ExpressionRendererProps>;
 
-export const createRenderer = (run: ExpressionRunner): ExpressionRenderer => ({
+export const createRenderer = (loader: IExpressionLoader): ExpressionRenderer => ({
   expression,
   onRenderFailure,
   ...options
 }: ExpressionRendererProps) => {
   const mountpoint: React.MutableRefObject<null | HTMLDivElement> = useRef(null);
 
+  const handlerRef = useRef((null as any) as ExpressionLoader);
```

Comment:
nit: can be written as `useRef<null | ExpressionLoader>(null);` for better type safety.